software/main.py

```diff
@@ -0,0 +1,266 @@
+                        #Follows where is ball so find_ball function is more efficient
+                    if delta<0:
+                        ball_side = 0
+                    else:
+                        ball_side = 1
+                    
+                        #SpeedY based on ball distance
+                    if dist > 175:
+                        speedY=0.2
+                    else:
+                        speedY=0.5
+                        #Controlls that balls location is ready for robot's orbit function
+                    if xcord < (reso_x_mid + 15) and xcord >(reso_x_mid -15) and dist > 250:
+                        #robot.center_ball(xcord)
+                        state="orbit"
+                    else:
+                        #Moves to ball
+                        speedR = delta*(-1.75)
+                        speedX = delta*0.25
+                        robot.move(speedX, speedR, speedY)
+                else:
+                    #If there is no ball
+                    state="find_ball"
+                        
+            elif state=="orbit":
+                print("--------Centering ball and basket--------")
+                
+                
+                #If there is ball
+                if len(processedData.balls)>0:
+                    processedData.balls.sort(key= lambda x: x.size)
+                    targeted_ball=processedData.balls[-1]
+                    xcord=targeted_ball.x
+                    ycord=targeted_ball.y
+                    dist=targeted_ball.distance
+                    delta=((xcord-reso_x_mid)/reso_x_mid)
+
+                    if (xcord > 434):
+                        rotate_dir = -2
+                    elif (xcord < 414):
+                        rotate_dir = 2
+                    else:
+                        rotate_dir = 0
+
+                    robot.move(rotate_dir, rotate_dir, rotate_dir)
+                    #controlls basket colour
+
+                    if basket_color == "magenta":
+                        basket = processedData.basket_m
+                    elif basket_color == "blue":
+                        basket = processedData.basket_b
+
+                    #if that kind of basket is in our list
+                    if basket.exists:
+                        #deltabasket=((basket.x-reso_x_mid)/reso_x_mid)
+                        if(abs(basket.x-reso_x_mid)<10):
+                            robot.stop()
+                            state="throw"
+                            print("Robot is ready to throw")
+                            #state="find_ball"
+                            
+                        elif abs(basket.x-reso_x_mid)>30:
+                            #orbit_speeds = Orbit(radius, reso_x_mid, xcord, dist)
+                             
+                            speedY=0
+                            speedX=0.5
+                            speedR=1000*speedX/(640-radius)
+
+                            if radius > (dist-5):

```

Comment:
The same statement is executed for both cases, why not use `if X or Y:`

Reply:
Done